src/server.py

```diff
@@ -27,13 +28,13 @@ class HackSoundPlayer(GObject.Object):
         'error': (GObject.SignalFlags.RUN_FIRST, None, (GLib.Error, str))
     }
 
-    def __init__(self, bus_name, sound_event_id, uuid_, metadata,
+    def __init__(self, sound_event_id, uuid_, metadata,
                  metadata_extras=None):
         super().__init__()
         self.logger = Logger(PlayerFormatter, self)
         # The following attributes (bus_name, sound_event_id and uuid) are used
         # internally by the logger to format the log messages.
-        self.bus_name = bus_name
+        self.bus_names = SortedSet([])
```

Comment:
II was using a SortedSet because I wrongly thought it kept the insertion order. I wanted to keep insertion order so bus names in the log are not "randomly" ordered on each log line. It was optimal to me to keep the order on which the bus names were added to that HackSoundPlayer object. Also I was using a set instead of a list since a set would allow to lookup for elements at an O(1) complexity (which should be useful to avoid adding a duplicate bus name or removing a bus name).

Unfortunately, the `set` type does not keep insertion order. But dicts do, so I could write my own OrderedSet class just for this purpose with something like:

```
class OrderedSet:
    def __init__(self):
        self._container = {}

    def add(self, item):
        self._container[item] = None

    def remove(self, item):
        del self._container[item]

    def __iter__(self):
        return iter(self._container.keys())
```

But also it is likely that there will not be too much items to iterate through.

Reply:
I think ordering the bus names by insertion order in the log is not worth adding the extra dependency or using a custom data structure. It's a rare case. I would prefer to use a normal Python `set`.